source/modulo_new_core/test/cpp_tests/communication/test_publisher_handler.cpp

```diff
@@ -0,0 +1,77 @@
+#include <gtest/gtest.h>
+
+#include <rclcpp/node.hpp>
+
+#include "modulo_new_core/communication/PublisherHandler.hpp"
+#include "modulo_new_core/communication/MessagePair.hpp"
+#include "modulo_new_core/exceptions/NullPointerException.hpp"
+
+using namespace modulo_new_core::communication;
+
+template<typename MsgT, typename DataT>
+static void test_publisher_interface(const std::shared_ptr<rclcpp::Node>& node, const DataT& value) {
+  // create message pair
+  auto data = std::make_shared<DataT>(value);
+  auto msg_pair = std::make_shared<MessagePair<MsgT, DataT>>(data, node->get_clock());
+
+  // create publisher handler
+  auto publisher = node->create_publisher<MsgT>("topic", 10);
+  auto publisher_handler =
+      std::make_shared<PublisherHandler<rclcpp::Publisher<MsgT>, MsgT>>(PublisherType::PUBLISHER, publisher);
+
+  // use in publisher interface
+  std::shared_ptr<PublisherInterface> publisher_interface(publisher_handler);
+  EXPECT_THROW(publisher_interface->publish(), modulo_new_core::exceptions::NullPointerException);
+  publisher_interface->set_message_pair(msg_pair);
+  EXPECT_NO_THROW(publisher_interface->publish());
+
+  publisher_interface = create_publisher_interface(publisher_handler, msg_pair);
+  EXPECT_NO_THROW(publisher_interface->publish());
+}
+
+class PublisherTest : public ::testing::Test {
+protected:
+  void SetUp() {
+    rclcpp::init(0, nullptr);
```

Comment:
For ROS context init and shutdown it may be preferable to use `static void SetUpTestSuite()` and `static void TearDownTestSuite()`. Test objects like the Node should still be constructed on each test case with `SetUp` for same initial conditions.

But, you only have one test case for now so it doesn't really matter.